[DSM] DDP-7172: LiquiBase for user_profile and other changes

pegahtah opened the pull request.

pegahtah committed:
created a new pom for ddp-database and added it in the modules and as a dependency for dsm-core and dsm-server

created the new changelog file and new changesets for creating the needed changes in the dss db

changed DSMServer to try and run the liquibase on DSS

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Created the new JDBI class `JdbiUser` and changed UserDao to use `withTxn` from `TransactionWrapper`
Changed `TransactionWrapper` to have a default database for cases of having 2 databases initialized

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

# Conflicts:
#	pepper-apis/dsm-core/pom.xml

SimoneMaiwald commented on pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java:
```diff
@@ -77,7 +79,8 @@ public static synchronized boolean isInitialized() {
         HOUSEKEEPING(ConfigFile.HOUSEKEEPING_DB_URL, ConfigFile.HOUSEKEEPING_NUM_POOLED_CONNECTIONS),
-        DSM(ConfigFile.DB_URL, ConfigFile.NUM_POOLED_CONNECTIONS);
+        DSM(ConfigFile.DB_URL, ConfigFile.NUM_POOLED_CONNECTIONS),
+        DDP_DB(ConfigFile.DSS_DB_URL, ConfigFile.NUM_POOLED_CONNECTIONS);
```
can we rename `DDP_DB` to `DSS_DB`

SimoneMaiwald replied:
can we rename `DDP_DB` to `SHARED_DB`

SimoneMaiwald commented on pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/util/LiquibaseUtil.java:
```diff
@@ -159,7 +166,8 @@ private void runMigrations(String changelogFile) throws LiquibaseException, SQLE
             throw originalError;
-        } finally {
+        }
+        finally {
```
is that checkstyle conform?

SimoneMaiwald commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java:
```diff
@@ -487,16 +487,37 @@ protected void setupDB(@NonNull Config config) {
         int maxConnections = config.getInt("portal.maxConnections");
         String dbUrl = config.getString("portal.dbUrl");
+        String dssDBUrl = config.getString("dss.dbUrl");
+        int dssMaxConnections = config.getInt("portal.maxConnections");//todo pegah add to vault
```
why add to vault? maybe separate maxConnections for dss? dss.maxConnections

pegahtah replied:
sorry, I meant secret manager  instead of vault.

SimoneMaiwald commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/user/JdbiUser.java:
```diff
@@ -0,0 +1,36 @@
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+public interface JdbiUser extends SqlObject {
```
why are we adding a class to get users from the old db

pegahtah replied:
Now that we are connecting to 2  different DBs, I made this class  to specify which DB I want to connect to. I have made the branch `DDP-7174` off of this branch, in which this class is changed to using new schema and new classes

SimoneMaiwald commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/user/UserDao.java:
```diff
@@ -1,117 +1,63 @@
-
     public Optional<UserDto> getUserByEmail(@NonNull String email) {
-        SimpleResult results = inTransaction((conn) -> {
+        SimpleResult results = TransactionWrapper.withTxn(TransactionWrapper.DB.DSM, handle -> {
```
should the transaction now point to dss?

pegahtah replied:
Will be changed in next branch ( it is in another ticket)

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
DDP-7172 needed changes to the liquidbase to get it working

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
DDP-7172 PR comments

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

# Conflicts:
#	pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/util/LiquibaseUtil.java

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
changing the length of `phone` column

pegahtah commented on pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/constants/ConfigFile.java:
```diff
@@ -6,6 +6,7 @@ public class ConfigFile {
 
     // field that specifies the url to the database
     public static final String DB_URL = "dbUrl";
+    public static final String DSS_DB_URL = "dssDbUrl";
```
Used for shared DB

pegahtah committed:
removed todo

pegahtah commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java:
```diff
@@ -488,16 +488,37 @@ protected void setupDB(@NonNull Config config) {
 
         logger.info("DB setup complete.");
+
+        TransactionWrapper.setDefaultDB(TransactionWrapper.DB.DSM);
```
Setting a default database for DSM so all our old classes and queries that use DSM DB will work

pegahtah committed:
changes for different role names being possible

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

gmakharat commented on pepper-apis/ddp-database/src/main/resources/db-changes/schema/DDP-7172_Authentication_tables.xml:
```diff
@@ -0,0 +1,94 @@
+                                 referencedColumnNames="role_id"/>
+    </changeSet>
+    <changeSet id="DDP-7172_user_settings_table" author="pegah">
+        <createTable tableName="user_settings">
```
this will replace `user_settings` of current DSM?

pegahtah replied:
yes! moving into DSS db

gmakharat commented on pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java:
```diff
@@ -488,16 +488,37 @@ protected void setupDB(@NonNull Config config) {
 
         int maxConnections = config.getInt("portal.maxConnections");
         String dbUrl = config.getString("portal.dbUrl");
+        String dssDBUrl = config.getString("dss.dbUrl");
```
I would extract `DSS` and `DSM` db connection openings into different methods with meaningful names, in better case I think each of it deserves its own class.
Some kind of hierarchy, like `DBConnector` as base abstract class and its two subclass `DSMConnector` and `DSSConnector`, we would have template method design pattern which would wrap the same steps like `try/catch` with its own content and each class would have constructor in which they would get arguments like `dbUrl, `maxConnections`, information which differs for each database.

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
checkstyle fix

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

# Conflicts:
#	pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/util/LiquibaseUtil.java

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into DDP-7172-LiquiBase_User_Table

pegahtah committed:
[DSM] DDP-7174: DSM_use_role_migration (#1545)

* DDP-7174
* Changed UserUtil.java and any class that was using it
* Created two Jdbi classes, `JdbiUser.java` and `JdbiUserRole.java` for new Database handling
* Changed `AuthenticationRoute.java` to use the guid from the db for the claims

* added todo

* creating AssigneeDto and removing Assignee.java

* participant list loading working with new db

* Added Daos and Dtos and Jdbi classes for User, UserRoleDao, UserPermissions and UserSettingsDao

Changed other classes to use the shared db

* added code for migrating roles and users

* inserting auth0 user id when missing
changing UserSettingsDao
changing user id from int to long
adding study_guid to ddp_instance queries

* Merging develop

* checkstyle fixes

* checkstyle fixes

* [DSM] DDP-7234: user_changes (#1612)

* DDP-7234 adding users

* DDP-7234 adding user_add permission
DDP-7234 adding new_user permissions

* DDP-7546 `GetRoleRoute` and relevant classes added

* changes to Dtos for json names

* DDP-7235 user access permission

* DDP-7235 assigning new roles to user

* DDP-7238 liquibase

* DDP-7234 changes to match the workflow of thee design

* DDP-7235
*removed new user role insertion query in `ddp-database`
*changed the `modify` functions to assign roles or change `user_profile` info

* DDP-7238
* adding `user_delete` permissions
* adding `DeactivateUserRoute.java` for deletion of user
* changed `User` and `Role` related classes to check for user being active
DDP-7235
*changes to `ModifyUserRoute` class to work with the designs

* DP-7234 Adding `study_admin` liquibase and instance role and role for studies that have phone numbers

* Merging develop

* checkstyle fixes

* checkstyle fixes

* DDP-7234 PR comments to use more Lombok annotations applied

* [DSM] DDP-7883_copy_dsm_user_id (#1641)

* DP-7238 Assignee table changes

* DP-7238 KitType changes to not use old access tables

* DP-7238 `ParticipantExit` changes to not use old access tables and get user names after retrieving exited pts

* DP-7238

* `SurveyTrigger` changes to not use old access tables and get user names after retrieving surveys
* Creating new method to return list of DSM users whether active or not

* DP-7238

* `AbstractionActivity` changes to not use old access tables and get user names after getting abstractions

* DP-7238

* `selectAllowedRealmsStartsLike` never used

* DP-7238

* change `KitDiscard` class for setting the usernames and also the `exitedKits` query in `application.conf`

* DP-7238

* change `ParticipantEvent` class for setting the usernames and also the `selectParticipantEvents` query in `application.conf`

* DDP-7883 removing queries from application.conf to classes

* checkstyle fixes

* DDP-7883 changing order to work with AllArgsConstructor
